x-pack/plugins/cloud/public/plugin.ts

```diff
@@ -135,4 +157,68 @@ export class CloudPlugin implements Plugin<CloudSetup> {
     const user = await security.authc.getCurrentUser().catch(() => null);
     return user?.roles.includes('superuser') ?? true;
   }
+
+  private async setupFullstory({
+    basePath,
+    security,
+  }: CloudSetupDependencies & { basePath: IBasePath }) {
+    const { enabled, org_id: orgId } = this.config.full_story;
+    if (!enabled || !orgId) {
+      return;
+    }
+
+    // Keep this import async so that we do not load any FullStory code into the browser when it is disabled.
+    const { initializeFullStory } = await import('./fullstory');
+    const userIdPromise: Promise<string | undefined> = security
+      ? loadFullStoryUserId({ getCurrentUser: security.authc.getCurrentUser })
+      : Promise.resolve(undefined);
+
+    initializeFullStory({
+      basePath,
+      orgId,
+      packageInfo: this.initializerContext.env.packageInfo,
+      userIdPromise,
+    });
+  }
 }
+
+/** @internal exported for testing */
+export const loadFullStoryUserId = async ({
+  getCurrentUser,
+}: {
+  getCurrentUser: () => Promise<AuthenticatedUser>;
+}) => {
+  try {
+    const currentUser = await getCurrentUser().catch(() => undefined);
+    if (!currentUser) {
+      return undefined;
+    }
+
+    const metadata: any = currentUser.metadata ?? {};
+    const cloudUserIdArray = metadata[ELASTIC_CLOUD_PRINCIPAL_ATTR_NAME];
+
+    // Log very defensively here so we can debug this easily if it breaks
+    if (
+      // Only proceed if the key is specified
```

Comment:
nit: I read the code comments as proceeding along through the 'happy' path, ultimately returning the `cloudUserId`. 
I'm not sure about other folks, but I find that it's easier to digest complex conditionals when they're extracted into a function with a very descriptive name. e.g. `isCloudUserIdValid`. Then we could do:
```
if (!isCloudUserIdValid) console.debug(.....)
return (cloudUserIdArray[0] as string | number).toString()
```